src/SharpCompress/Utility.cs

```diff
@@ -277,8 +277,33 @@ public static long TransferTo(this Stream source, Stream destination)
             long total = 0;
             while (ReadTransferBlock(source, array, out var count))
             {
+                destination.Write(array, 0, count);
                 total += count;
+            }
+            return total;
+        }
+        finally
+        {
+            ArrayPool<byte>.Shared.Return(array);
+        }
+    }
+
+    public static long TransferTo(this Stream source, Stream destination, int size)
+    {
+        var array = GetTransferByteArray();
+        try
+        {
+            long total = 0;
+            var remaining = size;
+            while (ReadTransferBlock(source, array, remaining, out var count))
+            {
                 destination.Write(array, 0, count);
+                total += count;
+                remaining -= count;
+                if (remaining - count < 0)
```

Comment:
This should be before the remaining -= count;
For exemple, if we have a remaining 15 bytes, and a count of 10. It will put remaining to 5 after the write. But 5 < 10 so it will never read the last 5 bytes.

Reply:
done